src/sentry/issues/issue_velocity.py

```diff
@@ -185,21 +223,37 @@ def get_latest_threshold(project: Project) -> float:
     ]
     client = get_redis_client()
     cache_results = client.mget(keys)  # returns None if key is nonexistent
-    threshold, stale_date = cache_results[0], cache_results[1]
-    now = convert_date_to_int(datetime.utcnow())
-    if (stale_date and int(stale_date) < now) or stale_date is None or threshold is None:
+    threshold = cache_results[0]
+    stale_date = None
+    if cache_results[1] is not None:
+        # TODO(isabella): update this once the old format is no longer being used
+        try:
+            stale_date = datetime.strptime(cache_results[1], STRING_TO_DATETIME)
+        except ValueError:
+            logger.info(
+                "issue_velocity.get_latest_threshold.old_date_format",
+                extra={"org_id": project.organization.id, "project_id": project.id},
```

Comment:
OK, so the more I thought about it, less I'm convinced that this will work as expected. 

This should use some kind of key versioning, because what's stored in the Redis will be forwards-compatible, but is not backward-compatible. Which means that for example during canary you'll have new code setting this key, old code reading it and blowing up (unless I'm missing something?).

IMO the way to go is is something like:
```
LEGACY_STALE_DATE_KEY = "new-issue-escalation-threshold-stale-date:{project_id}"
STALE_DATE_KEY = "new-issue-escalation-threshold-stale-date:v2:{project_id}"
⋮
keys = [
    THRESHOLD_KEY.format(project_id=project.id),
    STALE_DATE_KEY.format(project_id=project.id),
    LEGACY_STALE_DATE_KEY.format(project_id=project.id),
]
⋮   
cache_results = client.mget(keys)
⋮
if cache_results[1] is not None:
    # deal with new format
elif cache_results[2] is not None:
    # deal with old format
```